alacritty/src/renderer/platform.rs

```diff
@@ -0,0 +1,111 @@
+use std::num::NonZeroU32;
+
+use glutin::config::{ColorBufferType, Config, ConfigTemplateBuilder, GetGlConfig};
+use glutin::context::{ContextApi, ContextAttributesBuilder, NotCurrentContext, Version};
+use glutin::display::{Display, DisplayApiPreference, GetGlDisplay};
+use glutin::error::Result as GlutinResult;
+use glutin::prelude::*;
+use glutin::surface::{Surface, SurfaceAttributesBuilder, WindowSurface};
+
+use raw_window_handle::{RawDisplayHandle, RawWindowHandle};
+use winit::dpi::PhysicalSize;
+#[cfg(all(feature = "x11", not(any(target_os = "macos", windows))))]
+use winit::platform::unix;
+
+/// Create the GL display.
+pub fn create_gl_display(
+    raw_display_handle: RawDisplayHandle,
+    _raw_window_handle: Option<RawWindowHandle>,
+) -> GlutinResult<Display> {
+    #[cfg(target_os = "macos")]
+    let preference = DisplayApiPreference::Cgl;
+
+    #[cfg(windows)]
+    let preference = DisplayApiPreference::Wgl(Some(_raw_window_handle.unwrap()));
+
+    #[cfg(all(feature = "x11", not(any(target_os = "macos", windows))))]
+    let preference = DisplayApiPreference::GlxThenEgl(Box::new(unix::register_xlib_error_hook));
+
+    #[cfg(all(not(feature = "x11"), not(any(target_os = "macos", windows))))]
+    let preference = DisplayApiPreference::Egl;
+
+    unsafe { Display::from_raw(raw_display_handle, preference) }
+}
+
+pub fn pick_gl_config(
+    gl_display: &Display,
+    raw_window_handle: Option<RawWindowHandle>,
+) -> Result<Config, String> {
+    let mut default_config = ConfigTemplateBuilder::new().with_transparency(true);
+
+    if let Some(raw_window_handle) = raw_window_handle {
+        default_config = default_config.compatible_with_native_window(raw_window_handle);
+    }
+
+    let config_10bit = default_config
+        .clone()
+        .with_buffer_type(ColorBufferType::Rgb { r_size: 10, g_size: 10, b_size: 10 })
+        .with_alpha_size(2);
+
+    let configs = [
+        default_config.clone(),
+        config_10bit.clone(),
+        default_config.with_transparency(false),
+        config_10bit.with_transparency(false),
+    ];
+
+    for config in configs {
+        let gl_config = unsafe {
+            gl_display.find_configs(config.build()).ok().and_then(|mut configs| configs.next())
+        };
+
+        if let Some(gl_config) = gl_config {
+            return Ok(gl_config);
+        }
+    }
+
+    Err(String::from("failed to find suitable GL configuration."))
+}
+
+pub fn create_gl_context(
+    gl_display: &Display,
+    gl_config: &Config,
+    raw_window_handle: Option<RawWindowHandle>,
+) -> GlutinResult<NotCurrentContext> {
+    let context_attributes = ContextAttributesBuilder::new()
+        .with_context_api(ContextApi::OpenGl(Some(Version::new(3, 3))))
+        .build(raw_window_handle);
+
+    unsafe {
+        if let Ok(gl_context) = gl_display.create_context(gl_config, &context_attributes) {
+            Ok(gl_context)
+        } else {
+            // Fallback.
```

Comment:
This comment provides no value. We know it's a fallback since it's in the `else` branch of an error handler. If you want to keep this comment, you should provide more context.